core/invoker/src/main/scala/whisk/core/containerpool/ContainerProxy.scala

```diff
@@ -369,10 +369,16 @@ class ContainerProxy(factory: (TransactionId, String, ImageName, Boolean, ByteSi
       }
       .flatMap { activation =>
         val start = tid.started(this, LoggingMarkers.INVOKER_COLLECT_LOGS)
-        collectLogs(tid, container, job.action).map { logs =>
-          tid.finished(this, start)
-          activation.withLogs(logs)
-        }
+        collectLogs(tid, container, job.action)
+          .map { logs =>
+            tid.finished(this, start)
+            activation.withLogs(logs)
+          }
+          .recover {
+            case _ =>
+              tid.failed(this, start)
+              activation.withLogs(ActivationLogs(Vector(Messages.logFailure)))
+          }
```

Comment:
Thanks for adding the recovery path. If we end up in recovery: did we emit the error to the invoker log anywhere after the error occurred? Otherwise we have no change to understand what happened and don't even notice that activation logs cannot be processed.